New Bicep Release v0.13.1 Causing PSRule to fail. Error `template parameter xxxx does not use the required format`

**Description of the issue**

When a bicep template is compiled into an ARM template (.json) using the latest bicep release v0.13.1, PSRule cannot parse the compiled ARM .json template and complained certain parameters are not in the required format

**To Reproduce**

Steps to reproduce the issue:
i.e. when using the [SQL Database CARML module](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Sql/servers/databases/deploy.bicep) , the database Sku cannot be calculated by PSRule.


**Expected behaviour**

PSRule should successfully scan the compiled ARM template. PSRule **CAN** successfully scan it if the ARM template is compiled using previous bicep version `v0.12.40`.

**Error output**

[ERROR] Unable to expand resources because the template 'main2.json' was not valid with parameters 'main2.parameters.json'. The template parameter 'skuTier' does not use the required format

```text

```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.22.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Darwin 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Additional context**

I cannot publicly share the templates I used to reproduce the error, but I have emailed @BernieWhite the templates

--- a/docs/CHANGELOG-v1.md
+++ b/docs/CHANGELOG-v1.md
@@ -24,6 +24,12 @@ See [upgrade notes][1] for helpful information when upgrading from previous vers
 
 ## Unreleased
 
+What's changed since v1.22.0:
+
+- Bug fixes:
+  - Fixes template parameter does not use the required format by @BernieWhite.
+    [#1930](https://github.com/Azure/PSRule.Rules.Azure/issues/1930)
+
 ## v1.22.0
 
 What's changed since v1.21.2:

--- a/src/PSRule.Rules.Azure/Data/Template/TemplateVisitor.cs
+++ b/src/PSRule.Rules.Azure/Data/Template/TemplateVisitor.cs
@@ -1213,6 +1213,9 @@ namespace PSRule.Rules.Azure.Data.Template
             {
                 foreach (var parameter in innerParameters.Properties())
                 {
+                    if (parameter.Value is JValue parameterValueExpression)
+                        parameter.Value = ResolveToken(context, ResolveVariable(context, parameterValueExpression));
+
                     if (parameter.Value is JObject parameterInner)
                     {
                         if (parameterInner.TryGetProperty(PROPERTY_VALUE, out JToken parameterValue))

--- a/tests/PSRule.Rules.Azure.Tests/Tests.Bicep.2.bicep
+++ b/tests/PSRule.Rules.Azure.Tests/Tests.Bicep.2.bicep
@@ -22,7 +22,7 @@ module storage1 './Tests.Bicep.1.bicep' = {
   name: 'storage1'
   params: {
     name: storageName1
-    location: location
+    location: !empty(location) ? location : resourceGroup().location
   }
 }
 

--- a/tests/PSRule.Rules.Azure.Tests/Tests.Bicep.2.json
+++ b/tests/PSRule.Rules.Azure.Tests/Tests.Bicep.2.json
@@ -4,8 +4,8 @@
   "metadata": {
     "_generator": {
       "name": "bicep",
-      "version": "0.4.1318.3566",
-      "templateHash": "13856958272754005555"
+      "version": "0.13.1.58284",
+      "templateHash": "1497947304986475936"
     }
   },
   "parameters": {
@@ -40,13 +40,13 @@
         "mode": "Incremental",
         "parameters": {
           "name": {
-            "value": "[format('{0}{1}', split(reference(resourceId('Microsoft.Resources/deployments', 'storage1')).outputs.id.value, '/')[8], reference(resourceId('Microsoft.Resources/deployments', 'storage1')).outputs.unique.value)]"
+            "value": "[format('{0}{1}', split(reference(resourceId('Microsoft.Resources/deployments', 'storage1'), '2020-10-01').outputs.id.value, '/')[8], reference(resourceId('Microsoft.Resources/deployments', 'storage1'), '2020-10-01').outputs.unique.value)]"
           },
           "location": {
             "value": "[parameters('location')]"
           },
           "tags": {
-            "value": "[reference(resourceId('Microsoft.Resources/deployments', 'storage1')).outputs.tags.value]"
+            "value": "[reference(resourceId('Microsoft.Resources/deployments', 'storage1'), '2020-10-01').outputs.tags.value]"
           }
         },
         "template": {
@@ -55,8 +55,8 @@
           "metadata": {
             "_generator": {
               "name": "bicep",
-              "version": "0.4.1318.3566",
-              "templateHash": "4093724470647525487"
+              "version": "0.13.1.58284",
+              "templateHash": "10683227264194439860"
             }
           },
           "parameters": {
@@ -151,14 +151,14 @@
             },
             "unique": {
               "type": "string",
-              "value": "[uniqueString(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('name'))).primaryEndpoints.blob)]",
+              "value": "[uniqueString(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('name')), '2021-06-01').primaryEndpoints.blob)]",
               "metadata": {
                 "description": "A unique string that is generated from the blob endpoint."
               }
             },
             "endpointNIC": {
               "type": "string",
-              "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', format('pe-{0}', parameters('name')))).networkInterfaces[0].id]",
+              "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', format('pe-{0}', parameters('name'))), '2020-03-01').networkInterfaces[0].id]",
               "metadata": {
                 "description": "The ID of the endpoint NIC."
               }
@@ -183,9 +183,7 @@
           "name": {
             "value": "[parameters('storageName1')]"
           },
-          "location": {
-            "value": "[parameters('location')]"
-          }
+          "location": "[if(not(empty(parameters('location'))), createObject('value', parameters('location')), createObject('value', resourceGroup().location))]"
         },
         "template": {
           "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
@@ -193,8 +191,8 @@
           "metadata": {
             "_generator": {
               "name": "bicep",
-              "version": "0.4.1318.3566",
-              "templateHash": "4093724470647525487"
+              "version": "0.13.1.58284",
+              "templateHash": "10683227264194439860"
             }
           },
           "parameters": {
@@ -289,14 +287,14 @@
             },
             "unique": {
               "type": "string",
-              "value": "[uniqueString(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('name'))).primaryEndpoints.blob)]",
+              "value": "[uniqueString(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('name')), '2021-06-01').primaryEndpoints.blob)]",
               "metadata": {
                 "description": "A unique string that is generated from the blob endpoint."
               }
             },
             "endpointNIC": {
               "type": "string",
-              "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', format('pe-{0}', parameters('name')))).networkInterfaces[0].id]",
+              "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', format('pe-{0}', parameters('name'))), '2020-03-01').networkInterfaces[0].id]",
               "metadata": {
                 "description": "The ID of the endpoint NIC."
               }
@@ -322,7 +320,7 @@
             "value": "[parameters('location')]"
           },
           "tags": {
-            "value": "[reference(resourceId('Microsoft.Resources/deployments', 'storage1')).outputs.tags.value]"
+            "value": "[reference(resourceId('Microsoft.Resources/deployments', 'storage1'), '2020-10-01').outputs.tags.value]"
           }
         },
         "template": {
@@ -331,8 +329,8 @@
           "metadata": {
             "_generator": {
               "name": "bicep",
-              "version": "0.4.1318.3566",
-              "templateHash": "4093724470647525487"
+              "version": "0.13.1.58284",
+              "templateHash": "10683227264194439860"
             }
           },
           "parameters": {
@@ -427,14 +425,14 @@
             },
             "unique": {
               "type": "string",
-              "value": "[uniqueString(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('name'))).primaryEndpoints.blob)]",
+              "value": "[uniqueString(reference(resourceId('Microsoft.Storage/storageAccounts', parameters('name')), '2021-06-01').primaryEndpoints.blob)]",
               "metadata": {
                 "description": "A unique string that is generated from the blob endpoint."
               }
             },
             "endpointNIC": {
               "type": "string",
-              "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', format('pe-{0}', parameters('name')))).networkInterfaces[0].id]",
+              "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', format('pe-{0}', parameters('name'))), '2020-03-01').networkInterfaces[0].id]",
               "metadata": {
                 "description": "The ID of the endpoint NIC."
               }

